pallets/collator-selection/src/lib.rs

```diff
@@ -375,7 +402,14 @@ pub mod pallet {
 				index,
 				<frame_system::Pallet<T>>::block_number(),
 			);
-			Some(Self::assemble_collators())
+			let candidates_len = Self::candidates().len();
+			let result = Some(Self::assemble_collators());
```

Comment:
you should not wrap it in `Some` here, then you need to unwrap in the line below, this is like making things harder for yourself for no reason 😄